### DISCUSSION TITLE
HTTPSConnectionPool Read timed out and issues using Collectionless

### DISCUSSION THREAD

**Discussion by zpz5HAU-tgc3fgw2xwr** [meisnate12/Plex-Meta-Manager#200] [Q&A]
I'm running into HTTPSConnectionPool Read times out errors quite often, usually this happens with the large collections such as the collectionless fix.

The error I'm getting is:
```
HTTPSConnectionPool(host='plex.hostname.com', port=443): Read timed out. (read timeout=60)
```
When this happens, very rarely can it recover. Usually Plex crashes completely and I have to restart it. It's possible this is only an issue with large libraries.

**EDIT 2021-04-11**:  The Collectionless issues I had have been fixed on the develop branch (v1.7.2)

I̶'̶m̶ ̶a̶l̶s̶o̶ ̶h̶a̶v̶i̶n̶g̶ ̶i̶s̶s̶u̶e̶s̶ ̶w̶i̶t̶h̶ ̶t̶h̶e̶ ̶C̶o̶l̶l̶e̶c̶t̶i̶o̶n̶l̶e̶s̶s̶ ̶f̶i̶x̶,̶ ̶a̶ ̶f̶e̶w̶ ̶t̶h̶i̶n̶g̶s̶ ̶t̶o̶ ̶n̶o̶t̶e̶ ̶s̶o̶ ̶f̶a̶r̶:̶
̶
̶-̶ ̶T̶h̶e̶ ̶l̶a̶b̶e̶l̶ ̶a̶t̶t̶r̶i̶b̶u̶t̶e̶ ̶d̶o̶e̶s̶n̶'̶t̶ ̶g̶e̶t̶ ̶a̶p̶p̶l̶i̶e̶d̶
̶-̶ ̶T̶h̶e̶ ̶s̶o̶r̶t̶_̶t̶i̶t̶l̶e̶ ̶a̶t̶t̶r̶i̶b̶u̶t̶e̶ ̶d̶o̶e̶s̶n̶'̶t̶ ̶g̶e̶t̶ ̶a̶p̶p̶l̶i̶e̶d̶
̶-̶ ̶T̶h̶e̶ ̶c̶o̶l̶l̶e̶c̶t̶i̶o̶n̶_̶o̶r̶d̶e̶r̶ ̶a̶t̶t̶r̶i̶b̶u̶t̶e̶ ̶d̶o̶e̶s̶n̶'̶t̶ ̶g̶e̶t̶ ̶a̶p̶p̶l̶i̶e̶d̶
̶-̶ ̶T̶h̶e̶ ̶e̶x̶c̶l̶u̶d̶e̶_̶p̶r̶e̶f̶i̶x̶ ̶o̶p̶t̶i̶o̶n̶ ̶~̶ ̶n̶e̶e̶d̶s̶ ̶t̶o̶ ̶b̶e̶ ̶i̶n̶ ̶q̶u̶o̶t̶e̶s̶ ̶t̶o̶ ̶w̶o̶r̶k̶ ̶p̶r̶o̶p̶e̶r̶l̶y̶ ̶(̶t̶h̶e̶ ̶w̶i̶k̶i̶ ̶d̶o̶c̶u̶m̶e̶n̶t̶a̶t̶i̶o̶n̶ ̶d̶o̶e̶s̶n̶'̶t̶ ̶s̶h̶o̶w̶ ̶t̶h̶a̶t̶ ̶t̶o̶ ̶b̶e̶ ̶t̶h̶e̶ ̶c̶a̶s̶e̶)̶
̶
̶H̶e̶r̶e̶'̶s̶ ̶m̶y̶ ̶Y̶M̶L̶ ̶e̶n̶t̶r̶y̶ ̶f̶o̶r̶ ̶t̶h̶a̶t̶
```
̶ ̶ ̶"̶C̶o̶l̶l̶e̶c̶t̶i̶o̶n̶l̶e̶s̶s̶"̶:̶
̶ ̶ ̶ ̶ ̶l̶a̶b̶e̶l̶:̶ ̶.̶h̶i̶d̶d̶e̶n̶
̶ ̶ ̶ ̶ ̶s̶o̶r̶t̶_̶t̶i̶t̶l̶e̶:̶ ̶"̶~̶C̶o̶l̶l̶e̶c̶t̶i̶o̶n̶l̶e̶s̶s̶"̶
̶ ̶ ̶ ̶ ̶c̶o̶l̶l̶e̶c̶t̶i̶o̶n̶_̶o̶r̶d̶e̶r̶:̶ ̶a̶l̶p̶h̶a̶
̶ ̶ ̶ ̶ ̶p̶l̶e̶x̶_̶c̶o̶l̶l̶e̶c̶t̶i̶o̶n̶l̶e̶s̶s̶:̶
̶ ̶ ̶ ̶ ̶ ̶ ̶e̶x̶c̶l̶u̶d̶e̶_̶p̶r̶e̶f̶i̶x̶:̶
̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶-̶ ̶+̶
̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶-̶ ̶"̶~̶"̶
```

**Comment by zpz5HAU-tgc3fgw2xwr** [comment]
> I'm running into HTTPSConnectionPool Read times out errors quite often, usually this happens with the large collections such as the collectionless fix.
> 
> The error I'm getting is:
> 
> ```
> HTTPSConnectionPool(host='plex.hostname.com', port=443): Read timed out. (read timeout=60)
> ```
> 
> When this happens, very rarely can it recover. Usually Plex crashes completely and I have to restart it. It's possible this is only an issue with large libraries.

After some more testing, I think the problem happens because Plex starts a library scan while the script is running, causing some sort of timeout. I don't think I've run into this issue after turning off automatic scans and lengthening the periodic scan interval.

This could probably be fixed with some config options.
One option to pause library scans while the script is running, turning them back on afterwards.
Another option to choose whether the script should interrupt any ongoing scans, or wait until they're finished.

Bonus if the script sees which library is being scanned and starts a different one first instead of idling.

**Reply by meisnate12** [reply]
this is interesting ill look into if i can control the auto-scans at all

**Reply by trentmaetzold** [reply]
Having the same issue. I disabled the automatic scans in Plex. Going to test Kometa run now.
